src/archive.rs

```diff
@@ -205,6 +231,14 @@ impl<'a> Archive<dyn Read + 'a> {
     }
 }
 
+impl<'a> Archive<dyn SeekRead + 'a> {
+    fn skip_with_seek(&self, amt: u64) -> io::Result<()> {
+        (&self.inner)
+            .seek(io::SeekFrom::Current(amt as i64))
```

Comment:
Could this use a fallible `i64::try_from(amt)` and return the error as an `io::Error` to catch out-of-bounds cases?